scene/main/node.cpp

```diff
@@ -1770,6 +1770,16 @@ void Node::replace_by(Node* p_node,bool p_keep_data) {
 		}
 	}
 
+	Node* current_scene = NULL;
+#ifdef TOOLS_ENABLED
+	if (get_tree()->is_editor_hint())
+		current_scene = get_tree()->get_edited_scene_root();
+	else
+#endif
+    current_scene = get_tree()->get_current_scene();
```

Comment:
i don't think this will work, this is a special case and must be treated as
such from the outside, just make sure it works like before and that signals
are kept

On Thu, Jun 9, 2016 at 11:33 AM, Ignacio Etcheverry <
notifications@github.com> wrote:

> In scene/main/node.cpp
> <https://github.com/godotengine/godot/pull/5058#discussion_r66452166>:
>
> > @@ -1770,6 +1770,16 @@ void Node::replace_by(Node* p_node,bool p_keep_data) {
> >  		}
> >  	}
> >
> > +	Node* current_scene = NULL;
> > +#ifdef TOOLS_ENABLED
> > +	if (get_tree()->is_editor_hint())
> > +		current_scene = get_tree()->get_edited_scene_root();
> > +	else
> > +#endif
> > +    current_scene = get_tree()->get_current_scene();
>
> 😞
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/godotengine/godot/pull/5058/files/86049fb1c1e72602e6185cdd7098ba838a863ffb#r66452166>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AF-Z208QuNscEFCwC2OqJ4Ns6Tm9rtPXks5qKCQ1gaJpZM4Iubr3>
> .
>